src/Functions/FunctionsConversion.cpp

```diff
@@ -3736,20 +3737,41 @@ class FunctionCast final : public IFunctionBase
             element_wrappers.reserve(to_names.size());
             to_reverse_index.reserve(from_names.size());
 
+            size_t num_from_fields = 0;
             for (size_t i = 0; i < to_names.size(); ++i)
             {
                 auto it = from_positions.find(to_names[i]);
                 if (it != from_positions.end())
                 {
                     element_wrappers.emplace_back(prepareUnpackDictionaries(from_element_types[it->second], to_element_types[i]));
                     to_reverse_index.emplace_back(it->second);
+                    ++num_from_fields;
                 }
                 else
                 {
                     element_wrappers.emplace_back();
                     to_reverse_index.emplace_back();
                 }
             }
+
+            bool strict_named_tuple_conversion = false;
```

Comment:
The comment before the if (`and doesn't exist in @to_type it will be filled by default values.`) needs to be updated